File: budgette-frontend/src/types/transaction.ts
export type TransactionType = 'SEND' | 'RECEIVE' | 'WITHDRAW' | 'DEPOSIT' | 'PAYMENT' | 'AIRTIME'

export interface Transaction {
  id: string
  accountId: string
  type: TransactionType
  amount: number
  fees: number
  balanceAfter: number
  counterpartyName: string
  counterpartyPhone: string
  description: string
  date: string
}

export interface DashboardData {
  totalBalance: number
  currency: string
  accounts: import('./account').Account[]
  recentTransactions: Transaction[]
}

File: src/main/java/com/budgette/backend/domain/model/Transaction.java
package com.budgette.backend.domain.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class Transaction {

    private String id;
    private String accountId;
    private TransactionType type;
    private BigDecimal amount;
    private BigDecimal fees;
    private BigDecimal balanceAfter;
    private String counterpartyName;
    private String counterpartyPhone;
    private String description;
    private LocalDateTime date;
    private String externalId;
}
